From d434acd1ca8d8ddb185e543f2e94b6e8cc10821a Mon Sep 17 00:00:00 2001
From: Alvin Lee <Alvin.Lee2@amd.com>
Date: Fri, 17 Jul 2026 09:44:15 -0400
Subject: [PATCH 11434/15333] drm/amd/display: Update BW bounding box
 unconditionally for DCN6

[Description]
DCN6 needs to update BW bounding box in hw init in order to propagate
the Alt-Ch aperture info into DML. In this case update unconditionally
on dcn6 HW init.

Reviewed-by: Dillon Varone <dillon.varone@amd.com>
Signed-off-by: Alvin Lee <Alvin.Lee2@amd.com>
Signed-off-by: Roman Li <roman.li@amd.com>
Tested-by: Dan Wheeler <daniel.wheeler@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/display/dc/hwss/dcn60/dcn60_hwseq.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/hwss/dcn60/dcn60_hwseq.c b/drivers/gpu/drm/amd/display/dc/hwss/dcn60/dcn60_hwseq.c
index 05239e2f086f..72c2d3ca52f6 100644
--- a/drivers/gpu/drm/amd/display/dc/hwss/dcn60/dcn60_hwseq.c
+++ b/drivers/gpu/drm/amd/display/dc/hwss/dcn60/dcn60_hwseq.c
@@ -548,7 +548,6 @@ void dcn60_init_hw(struct dc *dc)
 	unsigned int edp_num;
 	uint32_t backlight = MAX_BACKLIGHT_LEVEL;
 	uint32_t user_level = MAX_BACKLIGHT_LEVEL;
-	int current_dchub_ref_freq = 0;
 
 	if (dc->clk_mgr && dc->clk_mgr->funcs && dc->clk_mgr->funcs->init_clocks) {
 		dc->clk_mgr->funcs->init_clocks(dc->clk_mgr);
@@ -589,8 +588,6 @@ void dcn60_init_hw(struct dc *dc)
 				dc->ctx->dc_bios->fw_info.pll_info.crystal_frequency,
 				&res_pool->ref_clocks.dccg_ref_clock_inKhz);
 
-			current_dchub_ref_freq = res_pool->ref_clocks.dchub_ref_clock_inKhz / 1000;
-
 			(res_pool->hubbub->funcs->get_dchub_ref_freq)(res_pool->hubbub,
 				res_pool->ref_clocks.dccg_ref_clock_inKhz,
 				&res_pool->ref_clocks.dchub_ref_clock_inKhz);
@@ -764,9 +761,8 @@ void dcn60_init_hw(struct dc *dc)
 		dc->caps.dmub_caps.fams_ver = dc->ctx->dmub_srv->dmub->feature_caps.fw_assisted_mclk_switch_ver;
 		dc->debug.fams2_config.bits.enable &=
 			dc->caps.dmub_caps.fams_ver == dc->debug.fams_version.ver; // sw & fw fams versions must match for support
-		if ((!dc->debug.fams2_config.bits.enable && dc->res_pool->funcs->update_bw_bounding_box)
-			|| res_pool->ref_clocks.dchub_ref_clock_inKhz / 1000 != current_dchub_ref_freq) {
-			/* update bounding box if FAMS2 disabled, or if dchub clk has changed */
+		if (dc->res_pool->funcs->update_bw_bounding_box) {
+			/* For DCN6 re-update unconditionally to propagate Alt-Ch address info into DML */
 			if (dc->clk_mgr)
 				dc->res_pool->funcs->update_bw_bounding_box(dc, dc->clk_mgr->bw_params);
 		}
-- 
2.55.0

